src/Components/Web.JS/src/Rendering/FocusOnNavigate.ts

```diff
@@ -0,0 +1,99 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+import { domFunctions } from '../DomWrapper';
+import { EnhancedNavigationStartEvent, JSEventRegistry } from '../Services/JSEventRegistry';
+
+const customElementName = 'blazor-focus-on-navigate';
+let currentFocusOnNavigateElement: FocusOnNavigateElement | null = null;
+let allowFocusOnEnhancedLoad = false;
+
+export function enableFocusOnNavigate(jsEventRegistry: JSEventRegistry) {
+  customElements.define(customElementName, FocusOnNavigateElement);
+  jsEventRegistry.addEventListener('enhancednavigationstart', onEnhancedNavigationStart);
+  jsEventRegistry.addEventListener('enhancedload', onEnhancedLoad);
+  document.addEventListener('focusin', onFocusIn);
+  document.addEventListener('focusout', onFocusOut);
+
+  // Focus the element on the initial page load.
+  if (document.readyState === 'loading') {
+    // There may be streaming updates on the initial page load, so we want to allow
+    // those to add elements matching the active selector.
+    allowFocusOnEnhancedLoad = true;
+    document.addEventListener('DOMContentLoaded', afterInitialPageLoad);
```

Comment:
Not sure that it matters, but might be worth passing `{ once: true }` (see https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener#options)

Reply:
Done.